sherpa/astro/data.py

```diff
@@ -83,8 +84,149 @@ def _notice_resp(chans, arf, rmf):
             arf.notice(bin_mask)
 
 
+def validate_ogip_energy_ranges(rtype, label, elo, ehi, ethresh=None):
+    """Check the lo/hi values are > 0, handling common error case.
+
+    Several checks are made, to make sure the parameters follow
+    the OGIP standard. If the checks fail a DataErr is raised.
+    When ethresh is set, the case where the low-edge of the start bin
+    is zero is treated as a special case rather than an error.
+    If a replacement is made (i.e. the low edge is set to ethresh) then
+    a warning is displayed.
+
+    Parameters
+    ----------
+    rtype : {'ARF', 'RMF'}
```

Comment:
It looks like this argument is only used in the error/warning messages, which seems odd. Also, there is a lot of potential for refactoring, with the goal of reducing the complexity. My feelings are not too strong about it, at least until the time comes (if ever) when we have to add more validation criteria. So unless @DougBurke thinks we should follow up on this, and as long as we discuss the cases that are not currently covered by tests, I am fine with the current implementation.

Reply:
When you talk about refactoring, how would you approach it (not this function itself, but the functionality in the PR)? I did what I felt made most sense to me, but that doesn't mean there aren't better approaches out there (i.e. I did not spend a lot of time exploring options).

Reply:
Validating input files is an important functionality, one that I feel is currently fragmented or, conversely, too centralized in the back end methods, which are rather opaque and complex, and there is a lot of duplication. But a) this does not need to be addressed by this PR and b) the function introduced by this PR is complex but not too much, so as I said I don't feel strongly about it. We can refactor the next time we need to add validation rules, if ever. As to how, I'd create classes for each input file type, rmfs, arfs, but also maybe mission file types (e.g. swift vs chandra vs xmm), with a factory method/function that returns the correct class based on the header metadata, with such classes being independent of the back end. But again, this is too much to ask in the scope of this PR (or others we have reviewed and merged in the recent past), and we have more pressing priorities anyway.